Segmentation fault when omitting the GitHub repo revision

When writing an (invalid) workflow YAML like this (note the missing revision `@master` after `OskarStark/php-cs-fixer-ga`):
```yml
name: Continuous integration

on:
  push:
  pull_request:
  workflow_dispatch:

jobs:
  check:
    runs-on: ubuntu-latest
    steps:

      - uses: actions/checkout@v2

      - uses: OskarStark/php-cs-fixer-ga
```
act will run into a segmentation fault:
```
[Continuous integration/check] ⭐  Run OskarStark/php-cs-fixer-ga
[Continuous integration/check]   ☁  git clone 'https://github.com/OskarStark/php-cs-fixer-ga' # ref=
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x898d7f]

goroutine 34 [running]:
github.com/go-git/go-git/v5.(*Repository).ResolveRevision(0xc0005bc480, 0x0, 0x0, 0x2d, 0x0, 0x0)
	/home/runner/go/pkg/mod/github.com/go-git/go-git/v5@v5.1.0/repository.go:1520 +0x33f
github.com/nektos/act/pkg/common.NewGitCloneExecutor.func1(0xe2c1e0, 0xc00043a9f0, 0x0, 0x0)
	/home/runner/work/act/act/pkg/common/git.go:239 +0x430
github.com/nektos/act/pkg/common.Executor.Then.func1(0xe2c1e0, 0xc00043a9f0, 0x23, 0xbbede0)
	/home/runner/work/act/act/pkg/common/executor.go:133 +0x4c
github.com/nektos/act/pkg/common.Executor.Then.func1(0xe2c1e0, 0xc00043a9f0, 0xb, 0xc000399fe0)
	/home/runner/work/act/act/pkg/common/executor.go:133 +0x4c
github.com/nektos/act/pkg/runner.(*RunContext).newStepExecutor.func1(0xe2c1e0, 0xc00043a9f0, 0x0, 0x0)
	/home/runner/work/act/act/pkg/runner/run_context.go:209 +0x386
github.com/nektos/act/pkg/common.Executor.Then.func1(0xe2c1e0, 0xc00043a9f0, 0xbbede0, 0xe369a0)
	/home/runner/work/act/act/pkg/common/executor.go:146 +0x17a
github.com/nektos/act/pkg/common.Executor.Then.func1(0xe2c1e0, 0xc00043a9f0, 0x1, 0x138fd90)
	/home/runner/work/act/act/pkg/common/executor.go:133 +0x4c
github.com/nektos/act/pkg/common.Executor.If.func1(0xe2c1e0, 0xc00043a9f0, 0xc0003b9b40, 0x1c)
	/home/runner/work/act/act/pkg/common/executor.go:154 +0x6a
github.com/nektos/act/pkg/runner.(*runnerImpl).NewPlanExecutor.func1(0xe2c1e0, 0xc0003a09c0, 0xc00044c0c0, 0xc000390f70)
	/home/runner/work/act/act/pkg/runner/runner.go:78 +0x24b
github.com/nektos/act/pkg/common.Executor.ChannelError.func1(0xe2c1e0, 0xc0003a09c0, 0xc00044c0c0, 0x0)
	/home/runner/work/act/act/pkg/common/executor.go:125 +0x45
github.com/nektos/act/pkg/common.NewParallelExecutor.func1.1(0xc00044c080, 0xc0003d6120, 0xe2c1e0, 0xc0003a09c0)
	/home/runner/work/act/act/pkg/common/executor.go:101 +0x56
created by github.com/nektos/act/pkg/common.NewParallelExecutor.func1
	/home/runner/work/act/act/pkg/common/executor.go:100 +0xb7
```

GitHub actions will instead display a sensible error message:
> The workflow is not valid. .github/workflows/continuous-integration.yml (Line: 16, Col: 15): Expected format {org}/{repo}[/path]@ref. Actual 'OskarStark/php-cs-fixer-ga' Input string was not in a correct format.

This was the problem behind #213, and a proper error message could have prevented that issue.

--- a/pkg/runner/step_context.go
+++ b/pkg/runner/step_context.go
@@ -57,6 +57,15 @@ func (sc *StepContext) Executor() common.Executor {
 		)
 	case model.StepTypeUsesActionRemote:
 		remoteAction := newRemoteAction(step.Uses)
+		if remoteAction.Ref == "" {
+			// GitHub's document[^] describes:
+			// > We strongly recommend that you include the version of
+			// > the action you are using by specifying a Git ref, SHA, or Docker tag number.
+			// Actually, the workflow stops if there is the uses directive that hasn't @ref.
+			// [^]: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions
+			msg := fmt.Sprintf("Expected format {org}/{repo}[/path]@ref. Actual '%s' Input string was not in a correct format.", step.Uses)
+			return common.NewErrorExecutor(fmt.Errorf("%s", msg))
+		}
 		if remoteAction.IsCheckout() && rc.getGithubContext().isLocalCheckout(step) {
 			return func(ctx context.Context) error {
 				common.Logger(ctx).Debugf("Skipping actions/checkout")
@@ -398,8 +407,6 @@ func newRemoteAction(action string) *remoteAction {
 	ra := new(remoteAction)
 	ra.Org = matches[1]
 	ra.Repo = matches[2]
-	ra.Path = ""
-	ra.Ref = "master"
 	if len(matches) >= 5 {
 		ra.Path = matches[4]
 	}

